src/libsync/owncloudpropagator.cpp

```diff
@@ -939,6 +940,11 @@ void PropagateDirectory::slotSubJobsFinished(SyncFileItem::Status status)
     emit finished(status);
 }
 
+void PropagateDirectory::slotAbortFinished()
+{
+    emit abortFinished();
```

Comment:
abortFinished signal is being send after 5 seconds timeout regardless if jobs terminated in that time or not. I give jobs max 5 seconds to finish what they started.